### PR TITLE
Upgrade to JBehave 4.4

### PR DESCRIPTION
**Pull request by valfirst**
#### Checklist
- [X] [Sign Allure CLA][cla]

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java [hunk 1]
```diff
@@ -134,7 +134,7 @@ public void beforeExamples(final List<String> steps, final ExamplesTable table)
     }
 
     @Override
-    public void example(final Map<String, String> tableRow) {
```
Comment (baev) [inline allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java]: Good job, JBehave. Breaking changes in minor release 👏 😢

Comment (valfirst) [reply allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java]: they are not really breaking changes, the methods are deprecated, so to be on the safe side it's better to switch to non-deprecated API

Comment (baev) [reply allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java]: > they are not really breaking changes, the methods are deprecated, so to be on the safe side it's better to switch to non-deprecated API

I don't like this change then. Now people should upgrade to the latest JBehave to use upcoming `allure-jbehave:2.11.0`

Comment (valfirst) [reply allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java]: it's up to you: to upgrade or not to upgrade such dependencies, in theory minor version change should be fully safe for users, but who knows how much users are dependent on particular implementations

Comment (baev) [reply allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java]: It's ok to upgrade the dependency. But I'd like to continue using of deprecated API at least for a some time, to give users an opportunity use older JBehave versions with latest Allure integration for a while

Comment (valfirst) [reply allure-jbehave/src/main/java/io/qameta/allure/jbehave/AllureJbehave.java]: ok, for now I've updated PR to just bump JBehave version

**Comment by baev** [comment]
@valfirst thanks!

**Pull request merged**